plugins/module_utils/zos_mvs_raw.py

```diff
@@ -99,3 +110,299 @@ def __init__(self, rc, stdout, stderr):
+            path_name=path,
+            normal_disposition=disposition_normal,
+            conditional_disposition=disposition_abnormal,
+            path_mode=mode,
+            access_group=access_group,
+            status_group=status_group,
+            file_data=file_data_type,
+            record_length=record_length,
+            block_size=block_size,
+            record_format=record_format,
+        )
+
+
+class RawInputDefinition(InputDefinition):
+    """Wrapper around InputDefinition to contain information about
+    desired return contents.
+
+    Args:
+        InputDefinition (InputDefinition): Input DD data type to be used in a DDStatement.
+    """
+
+    def __init__(
+            self,
+            content="",
+            return_content=None,
+            tmphlq="",
+            **kwargs
+    ):
+        """Initialize RawInputDefinition
+
+        Args:
+            content (str): The content to write to temporary data set / stdin.
+            return_content (dict, optional): Determines how content should be returned to the user. Defaults to {}.
+        """
+        self.return_content = ReturnContent(**(return_content or {}))
+        super().__init__(
+            content=content,
+            tmphlq=tmphlq)
+
+
+class RawOutputDefinition(OutputDefinition):
+    """Wrapper around OutputDefinition to contain information about
+    desired return contents.
+
+    Args:
+        OutputDefinition (OutputDefinition): Output DD data type to be used in a DDStatement.
+    """
+
+    def __init__(
+            self,
+            return_content=None,
+            tmphlq="",
+            **kwargs
+    ):
+        """Initialize RawOutputDefinition
+
+        Args:
+            content (str): The content to write to temporary data set / stdin.
+            return_content (dict, optional): Determines how content should be returned to the user. Defaults to {}.
+        """
+        self.return_content = ReturnContent(**(return_content or {}))
+        super().__init__(
+            tmphlq=tmphlq
+        )
+
+
+class ReturnContent(object):
+    """Holds information about what type of content
+    should be returned for a particular DD, if any.
+
+    Args:
+        object (object): The most base type.
+    """
+
+    def __init__(self, type=None, src_encoding=None, response_encoding=None):
+        """Initialize ReturnContent
+
+        Args:
+            type (str, optional): The type of content to return.
+                    Defaults to None.
+            src_encoding (str, optional): The encoding of the data set or file on the z/OS system.
+                    Defaults to None.
+            response_encoding (str, optional): The encoding to use when returning the contents of the data set or file.
+                    Defaults to None.
+        """
+        self.type = type
+        self.src_encoding = src_encoding
+        self.response_encoding = response_encoding
+
+
+def data_set_exists(name, volumes=None):
```

Comment:
I'm curious if there is a reason we can't use the `dest_exists` from the DataSet utils

Reply:
Good catch, let me test with the DataSet utils